src/index.js

```diff
@@ -311,14 +313,18 @@ export function createRoutes(options) {
   };
 
   // Check for unknown params:
-  const unknownParams = _.difference(Object.keys(options),
-                          Object.keys(constraints));
-  if (unknownParams.length > 0) {
+  const optionsSet = new Set(Object.keys(options));
+  const constraintsSet = new Set(Object.keys(constraints));
+  const unknownParams = new Set(
+    [...optionsSet].filter(opt => !constraintsSet.has(opt))
+  );
+
+  if (unknownParams.size > 0) {
     throw new Error('unknown parameters: ' + JSON.stringify(unknownParams));
   }
 
   // Merge default parameters into params:
-  const params = _.defaults(options, optionalParams);
+  const params = Object.assign({}, options, optionalParams);
```

Comment:
This is actually the inverse of what you want.
Lodash `_.defaults` applies objects from left to right, but prefers whichever property is assigned first. In contrast `Object.assign` prefers whichever property is assigned last.

If you reverse `options` and `optionalParams` you'll have the same behavior as intended.